File: src/com/anod/appwatcher/fragments/RemoveDialogFragment.java
package com.anod.appwatcher.fragments;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;

import com.anod.appwatcher.AppListContentProvider;
import com.anod.appwatcher.R;

public class RemoveDialogFragment extends DialogFragment {
	
    private static final String ARG_ROW_ID = "rowId";
	private static final String ARG_TITLE = "title";

	public static RemoveDialogFragment newInstance(String title, int rowId) {
    	RemoveDialogFragment frag = new RemoveDialogFragment();
        Bundle args = new Bundle();
        args.putString(ARG_TITLE, title);
        args.putInt(ARG_ROW_ID, rowId);
        frag.setArguments(args);
        return frag;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        String title = getArguments().getString(ARG_TITLE);
        final int rowId = getArguments().getInt(ARG_ROW_ID);
        String message = getString(R.string.alert_dialog_remove_message, title); 
        
        return new AlertDialog.Builder(getActivity())
             //   .setIcon(R.drawable.alert_dialog_icon)
                .setTitle(R.string.alert_dialog_remove_title)
                .setMessage(message)
                .setPositiveButton(R.string.alert_dialog_remove, 
                    new DialogInterface.OnClickListener() {
                        public void onClick(DialogInterface dialog, int whichButton) {
        					Uri deleteUri = AppListContentProvider.CONTENT_URI.buildUpon().appendPath(String.valueOf(rowId)).build();
        		            getActivity().getContentResolver().delete(deleteUri, null, null);
                        }
                    }
                )
                .setNegativeButton(R.string.alert_dialog_cancel,
                    new DialogInterface.OnClickListener() {
                        public void onClick(DialogInterface dialog, int whichButton) {
                            
                        }
                    }
                )
                .create();
    }
}

File: src/main/java/com/anod/appwatcher/fragments/RemoveDialogFragment.kt
package com.anod.appwatcher.fragments

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import com.anod.appwatcher.R
import com.anod.appwatcher.content.DbContentProviderClient
import com.anod.appwatcher.model.AppInfoMetadata
import com.anod.appwatcher.utils.ApplicationContext

class RemoveDialogFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val title = arguments.getString(ARG_TITLE)
        val rowId = arguments.getInt(ARG_ROW_ID)
        val message = getString(R.string.alert_dialog_remove_message, title)

        return AlertDialog.Builder(activity)
                .setTitle(R.string.alert_dialog_remove_title)
                .setMessage(message)
                .setPositiveButton(R.string.alert_dialog_remove) { _, _ ->
                    val cl = DbContentProviderClient(activity)
                    cl.updateStatus(rowId, AppInfoMetadata.STATUS_DELETED)
                    cl.close()
                    activity.finish()
                }
                .setNegativeButton(R.string.alert_dialog_cancel) { _, _ -> }
                .create()
    }

    companion object {
        private const val ARG_ROW_ID = "rowId"
        private const val ARG_TITLE = "title"

        fun newInstance(title: String, rowId: Int): RemoveDialogFragment {
            val frag = RemoveDialogFragment()
            val args = Bundle()
            args.putString(ARG_TITLE, title)
            args.putInt(ARG_ROW_ID, rowId)
            frag.arguments = args
            return frag
        }
    }
}
